src/hotspot/cpu/x86/stubGenerator_x86_64_aes.cpp

```diff
@@ -3066,235 +3402,138 @@ void StubGenerator::aesgcm_encrypt(Register in, Register len, Register ct, Regis
   const XMMRegister ZTMP14 = xmm20;
   const XMMRegister ZTMP15 = xmm21;
   const XMMRegister ZTMP16 = xmm30;
-  const XMMRegister COUNTER_INC_MASK = xmm18;
+  const XMMRegister ZTMP17 = xmm31;
+  const XMMRegister ZTMP18 = xmm1;
+  const XMMRegister ZTMP19 = xmm18;
+  const XMMRegister ZTMP20 = xmm8;
+  const XMMRegister ZTMP21 = xmm22;
+  const XMMRegister ZTMP22 = xmm23;
+  const XMMRegister ZTMP23 = xmm26;
+  const XMMRegister GH = xmm24;
+  const XMMRegister GL = xmm25;
+  const XMMRegister SHUF_MASK = xmm29;
+  const XMMRegister ADDBE_4x4 = xmm27;
+  const XMMRegister ADDBE_1234 = xmm28;
+  const XMMRegister ADD_1234 = xmm13;
+  const KRegister MASKREG = k1;
+  const Register pos = rax;
+  const Register rounds = r15;
+  const Register CTR_CHECK = r14;
+
+  const int stack_offset = 64;
+  const int ghashin_offset = 64;
+  const int aesout_offset = 64;
+  const int hashkey_offset = 0;
+  const int hashkey_gap = 16 * 32;
+  const int HashKey_32 = 0;
+  const int HashKey_16 = 16 * 16;
 
-  __ movl(pos, 0); // Total length processed
-  // Min data size processed = 768 bytes
-  __ cmpl(len, 768);
-  __ jcc(Assembler::less, ENC_DEC_DONE);
+  __ movl(pos, 0);
+  __ cmpl(len, 256);
+  __ jcc(Assembler::lessEqual, ENC_DEC_DONE);
```

Comment:
Could this handle be Assembler::less, in which case you can handle 256 byte length as well.